airflow-core/tests/unit/models/test_deadline.py

```diff
@@ -160,16 +161,37 @@ def test_repr_without_callback_kwargs(self, deadline_orm, dagrun, session):
         )
 
     @pytest.mark.db_test
-    def test_handle_miss_async_callback(self, dagrun, deadline_orm, session):
+    @pytest.mark.parametrize(
+        "kwargs",
+        [
+            pytest.param(TEST_CALLBACK_KWARGS, id="non-empty kwargs"),
+            pytest.param(None, id="null kwargs"),
+        ],
+    )
+    def test_handle_miss_async_callback(self, dagrun, session, kwargs):
+        deadline_orm = Deadline(
+            deadline_time=DEFAULT_DATE,
+            callback=AsyncCallback(TEST_CALLBACK_PATH, kwargs),
+            dagrun_id=dagrun.id,
+        )
+        session.add(deadline_orm)
+        session.flush()
         deadline_orm.handle_miss(session=session)
         session.flush()
 
         assert deadline_orm.trigger_id is not None
-
         trigger = session.query(Trigger).filter(Trigger.id == deadline_orm.trigger_id).one()
         assert trigger is not None
+
         assert trigger.kwargs["callback_path"] == TEST_CALLBACK_PATH
-        assert trigger.kwargs["callback_kwargs"] == TEST_CALLBACK_KWARGS
+
+        kwargs = trigger.kwargs["callback_kwargs"]
+        context = kwargs.pop("context")
+        assert kwargs == kwargs
```

Comment:
I don't think you meant to assign this to `kwargs` here, because L190 is not testing anything anymore, because of course `kwargs == kwargs` :)